### ISSUE TITLE
Finalize MVP Opportunity Page

### ISSUE THREAD

**Issue by MatthewVita** [saxifrage/learn#141]
work with @cameronscott137 to figure out what aspects of the opportunity page we need for MVP. ensure functionality, data, and HTML/CSS is in place.

**Comment by cameronscott137** [comment]
@MatthewVita Polished draft is in place on the opportunity_ui branch. Points to resolve:
- Plug in neighborhood and registration date variables.
- Reformat the Organization section to account for rectangular images.
- Plug in event contact variables.
- Determine a path forward with badges.
- Determine a path forward with attendees. Based on previous conversations with @timothyfcook, attendees will likely be shelved until v2, but certainly open to continuing that discussion.
- Make the individual event detail rows display conditionally if data is present. Will keep event pages from looking sparse. @MatthewVita, looks like I can leverage `ng-if` here?

**Comment by timothyfcook** [comment]
@cameronscott137 yeah, let's not include the Attendees in the opportunity page for now. We won't have learner accounts until after our initial launch. Once we finalize the design implementation and get our Organizer Accounts up and running we can launch!

**Comment by timothyfcook** [comment]
I'm fine with leaving the Badges section. Maybe for opportunities that don't have a badge (which will initially be most of them) we could put in a greyed-out placeholder badge to suggest that it should exist someday. I like the idea of subtly suggesting that Badges ought to be part of programs, even if they aren't yet.

**Comment by timothyfcook** [comment]
@MatthewVita let me know when this is implemented in the App so I can take a look at it.

**Comment by MatthewVita** [comment]
@cameronscott137 looks great! Yes, `ng-if` is the correct choice (it actually adds/removes elements from the DOM tree based on the conditional rather than simply hiding/showing).

@timothyfcook Cameron's work is currently in the `opportunity_ui`, but if you just want to "take a look at it":

![dev pittsburgh city as a campus opportunity instances](https://cloud.githubusercontent.com/assets/1715206/8611338/8f0dcf62-268d-11e5-8773-b3f856fc57e0.png)

## my actionables:
- [x] make dates "user friendly" via Angular filters
- [x] incorporate "neighborhood" into data model
- [x] figure out reasoning behind "register by" to determine next steps there
- [x] follow up on if opportunity instances are to have "short description" as well
- [x] implement hover over or click info on fake badges (waiting on Tim)

**Comment by MatthewVita** [comment]
@timothyfcook regarding badges for MVP... I say we have 2 grey badge "silhouettes" on each instance (no logic or associated data in the db... literally _just_ the static images).

If a user hovers/taps/clicks a badge, we'll just issue an ugly JavaScript `alert('badges are coming soon!')`

/$0.02

**Comment by timothyfcook** [comment]
I like this strategy. @cameronscott137 let's go ahead with two greyed-out/silhouette place-holder badges with that "Badges are coming soon!" hover-over.

**Comment by MatthewVita** [comment]
@cameronscott137 tackled date fmt (frontend) and neighborhood (backend) in my branch here: updated-opportunity-ui

![capture](https://cloud.githubusercontent.com/assets/1715206/8631422/94380334-2743-11e5-97d4-ad85602f8b34.PNG)

@timothyfcook:
- What is "register by"? Should I add it to our data model? Is it nullable?
- Are we still doing "short description" for opp. instances? Unless you have a valid reason, I think we can roll with just one description... right?
- regarding "Badges are coming soon! hover-over" ... this is super easy with the HTML `title` attribute, but hovers don't exist on mobile. Is this okay?

**Comment by MatthewVita** [comment]
pinging @timothyfcook

**Comment by timothyfcook** [comment]
@MatthewVita sorry, missed this. 
- "register by" is a registration deadline.
- just one description is fine.
- okay with no hovers on mobile.

**Comment by MatthewVita** [comment]
@timothyfcook thank you!

Here's what we have now:

![dev pittsburgh city as a campus opportunity instances](https://cloud.githubusercontent.com/assets/1715206/8887293/fb5615fe-3249-11e5-8664-b4bd65876f52.png)

**Comment by MatthewVita** [comment]
@timothyfcook anything else we want on this page for the MVP?

Some things that I'm thinking we could amend:
- Add CSS to the logo to make it have rounded corners/box shadow
- Widen opportunity instance title space so it doesn't wrap
- Comment out `contact` and `attendees` boxes for now
- Determine if we want `venue`, if so, I'll add it to rails
- Determine what `register now` button actually does? Sends an email via a prompt for entry or actually inserts a record into some sort of `attendees` join table
- Clean up `organization` box styling (looks a little off) (also change name to "organizer", right?)

**Comment by timothyfcook** [comment]
Thanks @MatthewVita ! 

Agreed on all fronts.
- Yes, we do want `venue` or `Location` works; it was part of the original [data spec](https://github.com/saxifrage/learn/blob/master/examples/opportunity_instance_linked_data.json) I wrote. 
- Yes it should be `organizer`
  "Register Now" should open a new tab and take you to the `registration_url`

If an event is no longer active (i.e. it is passed the `registration_deadline` data), then maybe the button could change to "Join Waiting List". This could be something that shows up in Organizer Admin profiles so they could see Demand building up for events that haven't run in a while.

Just heard from @cameronscott137 ...sounds like he should be back on the case this week.

**Comment by MatthewVita** [comment]
@timothyfcook this issue will live until those visual tweaks are in place. I've created a separate issue to add venue to the backend/frontend here https://github.com/saxifrage/learn/issues/152. I'm thinking this will be a good candidate for Tim S to take on since it's smaller, but involves small tweaks on the frontend and the backend.

@cameronscott137 's tasks
- [x] Clean up organization box styling (looks a little off)
- [x] Add CSS to the logo to make it have rounded corners/box shadow

My tasks
- [x] Widen opportunity instance title space so it doesn't wrap
- [x] Comment out contact and attendees boxes for now
- [x] Implement register now link action
- [x] Disable register now if past deadline [0]
- [x] Change "organization" to "organizer"

[0] In the spirit of MVP, let's just gray out the button if the deadline has past. We'll worry about advanced functionality later on

**Comment by cameronscott137** [comment]
@MatthewVita Organization box is all cleaned up and committed here: https://github.com/saxifrage/learn/commit/4a6cd4b0016cf5b5e5a1f066662d6a722c62a364

For my second to-do, which logo were you referring to? The organization logo on the sidebar?

**Comment by MatthewVita** [comment]
@cameronscott137 yes

**Comment by timothyfcook** [comment]
@chelseaerdner can you upload your original design for the Opportunity pages here? 

I like everything except maybe the event details section. Each type of data there should be expressed differently and the field names used only when necessary as a prompt.
![img_0292](https://cloud.githubusercontent.com/assets/1823300/9644006/f294f454-5192-11e5-8f13-d07ec2bf11e4.JPG)

**Comment by MatthewVita** [comment]
closing because of https://github.com/saxifrage/cityasacampus/issues/191
